### DISCUSSION TITLE
Creating a record using the resource class

### DISCUSSION THREAD

**Discussion by roqkabel** [filamentphp/filament#12170] [Help]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

How do I get to create a record using the resource class without using the Model.

Something like this:

```php
             StockAdjustmentResource::create([
                    "type" => "increment",
                     "model" => "PURCHASE_ORDER",
                      "product_id" => $purchase_order_item->product->id,
                       "previous_quantity" => $purchase_order_item->product->stock_quantity,
                       "quantity" => $purchase_order_item->quantity
              ]);
```

Reason :

I have already modified a beforeCreate method on the resource which does some checks and record creation.

And I don't want to repeat the same logic in a hook of an Edit Action or a different resource action.

I will be glad to get any trick or idea on how to do it.

**Comment by leandrocfe** [comment]
Maybe a trait?

**Reply by roqkabel** [reply]
Yes, an added feature to the resource class would be great.

But in the meantime, I need a work-around.
